src/server/coprocessor/endpoint.rs

```diff
@@ -485,14 +480,169 @@ fn get_chunk(chunks: &mut Vec<Chunk>) -> &mut Chunk {
     chunks.last_mut().unwrap()
 }
 
+struct SortRow {
+    key: Vec<Datum>,
+    handle: i64,
+    data: RowColsDict,
+    order_cols: Rc<Vec<ByItem>>,
+    ctx: Rc<EvalContext>,
+    err: Rc<RefCell<Option<String>>>,
+}
+
+
+impl SortRow {
+    fn new(handle: i64,
+           order_cols: Rc<Vec<ByItem>>,
+           ctx: Rc<EvalContext>,
+           data: RowColsDict,
+           key: Vec<Datum>,
+           err: Rc<RefCell<Option<String>>>)
+           -> SortRow {
+        SortRow {
+            key: key,
+            handle: handle,
+            data: data,
+            order_cols: order_cols,
+            ctx: ctx,
+            err: err,
+        }
+    }
+
+    fn cmp_and_check(&self, right: &SortRow) -> Result<CmpOrdering> {
+        // check err
+        try!(self.check_err());
```

Comment:
Error may caused by `into_sort_vec` which would use cmp. Sort would be finished  since we return equal when error happens. So here added err check to improve sort performance when error happens.